Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/workbench/RetrieveProjectManager.java

```diff
@@ -283,4 +283,26 @@ public int compare(IProject o1, IProject o2) {
         }       
         return projectList;
     }
+
+    /**
+     * This method retrieves the list of projects associated with the search Term and 
+     * accessibility of the project. 
+     * It uses the Project shallow mapper to give a {@link List} of {@link IProject} of domain type {@link ProjectProxy}.
+     * @param searchTerm - The search term which is a text, searched in name and description of the project.
+     * @param accessibility - accessibility of the project.
+     * @return List<IProject> - list of projects associated with the given searchTerm and accessibility of the project.
+     * @throws QuadrigaStorageException 
+     */
+    @Override
+    @Transactional
+    public List<IProject> getProjectListBySearchTermAndAccessiblity(String searchTerm, String accessibility) throws QuadrigaStorageException {
+    	List<ProjectDTO> projectDTOList = dbConnect.getAllProjectsDTOBySearchTermAndAccessiblity(searchTerm, accessibility);
+    	List<IProject> projectList = new ArrayList<IProject>();
+    	if(projectDTOList!=null) {
+    		for(ProjectDTO projectDTO : projectDTOList){
+    			projectList.add(projectShallowMapper.getProjectDetailsForSearch(projectDTO, searchTerm));

```

Comment:
just use the existing method to map projects here and put the logic you currently have in the mapper here.